jupyter_client/manager.py

```diff
@@ -272,6 +272,10 @@ def finish_shutdown(self, waittime=None, pollinterval=0.1):
             if self.is_alive():
                 time.sleep(pollinterval)
             else:
+                # If there's still a proc, wait and clear
+                if self.has_kernel:
+                    self.kernel.wait()
```

Comment:
Is there a reason to wait if is_alive() is False?

Reply:
Thanks @minrk . Since there's still a process object around, I consider it good practice and we're not seeing runtime issues in the sequence.